WordPress/gutenberg
# [Block: Navigation link] Fix PHP notice

## Description

Closes #34883.
Avoid checking for post status if the post ID does not exist.

Post ID's that does not exist can occur for example when a template with a navigation block is saved and exported and used on 
another website, or if someone edits the ID incorrectly.

## How has this been tested?




1. Create a new post or page. Open the code editor.
2. Copy & paste this example code for the navigation link. **The ID needs to be a post ID that does not exist.**
```



```
3. Save and view the front, there should be no PHP notice.
4. View the post or page in the site editor, there should be no PHP notice.

## Types of changes




Bug fix 

## Checklist:
